ihatemoney/web.py

```diff
@@ -898,6 +901,50 @@ def statistics():
     )
 
 
+def build_etag(project_id, last_modified):
+    return hashlib.md5(
+        (current_app.config["SECRET_KEY"] + project_id + last_modified).encode()
+    ).hexdigest()
+
+
+@main.route("/<project_id>/feed/<string:token>.xml")
+def feed(token):
+    verified_project_id = Project.verify_token(
+        token, token_type="feed", project_id=g.project.id
+    )
+    if verified_project_id != g.project.id:
+        abort(404)
+
+    weighted_bills = g.project.get_bill_weights_ordered().paginate(
+        per_page=100, error_out=True
+    )
+
+    bills_last_modified = [
+        bill.versions[0].transaction.issued_at for bill in g.project.get_bills()
+    ]
```

Comment:
Actually, since you only return the 100 most recent bills, you could also just check the last modified time of the 100 most recent bills. I suggest:

- fetch the 100 most recent bills (only once)
- use it to compute the last modified time
- use it again to feed it to the template (if needed according to the previous step)

Reply:
Done.